### PR TITLE
Sentry: Issue details and assign action

### PR DESCRIPTION
**Pull request by thomaspaulmann**
## Description

I adopted the new `@raycast/utils` to the Sentry extension. Furthermore I added a detail view and an action to assign issues.

## Screencast

sentry 2022-08-19 at 18 04 53

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by mattisssa** [comment]
Hm, I got into this screen:

CleanShot 2022-08-21 at 12 29 12@2x

```
Error: Worker terminated due to reaching memory limit: JS heap out of memory

__node_internal_captureLargerStackTrace:errors:464:5
NodeError:errors:371:5
Worker.[kOnExit]:worker:279:26
Worker.<computed>.onexit:worker:201:20
12:29:16.636 
Error: Worker terminated due to reaching memory limit: JS heap out of memory

__node_internal_captureLargerStackTrace:errors:464:5
NodeError:errors:371:5
Worker.[kOnExit]:worker:279:26
Worker.<computed>.onexit:worker:201:20
```

The Store version works fine tho.

File: extensions/sentry/src/sentry.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  });
+}
+
+export function useUsers(organizationSlug: string, projectId?: string) {
+  const searchParams = new URLSearchParams();
+  if (projectId) {
+    searchParams.append("project", projectId);
+  }
+
+  return useFetch<User[]>(`https://sentry.io/api/0/organizations/${organizationSlug}/users/?` + searchParams, {
```
Comment (sxn) [inline extensions/sentry/src/sentry.ts:37]: i -think- this introduces a new scope requirement. My old sentry API key, which has access to the scopes mentioned in the README (`event:read` and `project:read`) doesn't seem to work, or at least not fully:
image

I generated an API key that has access to `org:read`, too, and that one seems to work as it should.

**Comment by sxn** [comment]
not a must, but a custom [EmptyView](https://developers.raycast.com/api-reference/user-interface/list?q=emptyview#list.emptyview) could be a nice touch 😺

image

**Comment by thomaspaulmann** [comment]
> not a must, but a custom [EmptyView](https://developers.raycast.com/api-reference/user-interface/list?q=emptyview#list.emptyview) could be a nice touch 😺
> 
> image

I love custom empty views!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @thomaspaulmann 👋 

If using the wrong `Auth Token` I just get this view

CleanShot 2022-09-12 at 14 43 29@2x

It would be nice to get the loader to stop and show the `List.Emptyview` instead

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by thomaspaulmann** [comment]
sentry 2022-09-14 at 11 28 56

@pernielsentikaer I added a nice error screen and also updated the README with the new required scopes.

File: extensions/sentry/src/sentry.ts (hunk 1 above)
Comment (thomaspaulmann) [reply extensions/sentry/src/sentry.ts:37]: I explained the new scopes in the README and also added a proper view for the unauthorized state. I hope this is enough for a "smooth" transition.

**Comment by pernielsentikaer** [comment]
@thomaspaulmann did you push the changes yet?

**Comment by thomaspaulmann** [comment]
@pernielsentikaer something was off with my local setup. Pushed my changes!

**Review by pernielsentikaer — APPROVED** [review]
Looks good to me, approved 🔥

**Comment by thomaspaulmann** [comment]
The [docs](https://docs.sentry.io/api/auth/) state that this should also work with self-hosted instaces.

**Comment by NoorBayari** [comment]
Hi @thomaspaulmann, first of all, thank you for creating this extension - it's super useful! I just wanted to clarify that although tokens can be created on a self-hosted Sentry instance, they can't be used on sentry.io. One potential solution could be to add a feature that allows configuring the URL, so the extension can communicate with a self-hosted instance instead. Hope that helps!

**Comment by thomaspaulmann** [comment]
That should be possible. Can you send me some example cURL request that works for you so that I can make the changes. Unfortunately I don’t have access to a self-hosted instance of Sentry to test it.

**Pull request merged**